### PR TITLE
Remove test helpers

### PR DESCRIPTION
**Pull request by FK78**
Resolves #3264 

**Overall change:** _Exported @bbc/psammead-test-helpers into Simorgh._

**Code changes:**

- _Removed testHelpers.js._
- _Changed import statements to use @bbc/psammead-test-helpers._
- _Updated snapshots._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/testHelpers/loggerMock.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import nodeLogger from '../app/lib/logger.node';
+
+export const mocks = {
```
Comment (dr3) [inline src/testHelpers/loggerMock.js]:
```suggestion
const mocks = {
```

File: src/server/utilities/logResponseTime/index.test.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { loggerMock } from '../../../testHelpers/index';
+import { mocks } from '../../../testHelpers/loggerMock';
```
Comment (dr3) [inline src/server/utilities/logResponseTime/index.test.js]: Can you change these to `import loggerMock from`... everywhere please :)

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/components/IfAboveIE9Comment/__snapshots__/index.test.jsx.snap [hunk 3]
```diff
@@ -1,23 +1,7 @@
...
-  <div
-    dangerouslySetInnerHTML={
-      Object {
-        "__html": "<!--<![endif]-->",
-      }
-    }
-  />,
-]
+<div>
+  <!--[if !IE]&gt;&lt;!-->
```
Comment (dr3) [inline src/app/components/IfAboveIE9Comment/__snapshots__/index.test.jsx.snap]: Snapshot no longer reflects previous coverage, needs investigating

File: src/app/components/Metadata/helpers/__snapshots__/iconLinks.test.js.snap [hunk 4]
```diff
@@ -1,64 +1,9 @@
...
-    type="image/png"
-  />,
-  <link
-    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-192x192.png"
-    rel="icon"
-    sizes="192x192"
-    type="image/png"
-  />,
-]
+<link
```
Comment (dr3) [inline src/app/components/Metadata/helpers/__snapshots__/iconLinks.test.js.snap]: Snapshot no longer reflects previous coverage, needs investigating

File: src/app/components/ResourceHints/__snapshots__/index.test.jsx.snap [hunk 5]
```diff
@@ -1,24 +1,9 @@
...
-  <link
-    href="http://foo.com"
-    rel="dns-prefetch"
-  />,
-  <link
-    href="https://bar.com"
-    rel="dns-prefetch"
-  />,
-]
+<link
```
Comment (dr3) [inline src/app/components/ResourceHints/__snapshots__/index.test.jsx.snap]: Snapshot no longer reflects previous coverage, needs investigating

File: src/app/containers/ATIAnalytics/canonical/__snapshots__/index.test.jsx.snap [hunk 6]
```diff
@@ -1,17 +1,3 @@
...
-    style={
-      Object {
-        "position": "absolute",
-      }
-    }
-    width="1px"
-  />
-</noscript>
-`;
+exports[`Canonical ATI Analytics should render correctly 1`] = `<noscript />`;
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/canonical/__snapshots__/index.test.jsx.snap:5]: Snapshot no longer reflects previous coverage, needs investigating

File: src/app/containers/ArticleFigure/__snapshots__/index.test.jsx.snap [hunk 7]
```diff
@@ -92,40 +86,25 @@ exports[`ArticleFigure should render a lazyloaded image when lazyLoad set to tru
...
-            alt="Pauline Clayton"
-            className="c4"
-            height={null}
-            sizes={null}
-            src="https://ichef.bbci.co.uk/news/640/cpsprodpb/E7DB/production/_101655395_paulineclayton.jpg"
-            srcSet={null}
-            width={640}
-          />
-        </noscript>
+        <noscript />
```
Comment (dr3) [inline src/app/containers/ArticleFigure/__snapshots__/index.test.jsx.snap:103]: Snapshot no longer reflects previous coverage, needs investigating

File: src/app/containers/ImageWithPlaceholder/__snapshots__/index.test.jsx.snap [hunk 8]
```diff
@@ -42,38 +37,13 @@ exports[`ImageWithPlaceholder should render a lazyloaded image when lazyLoad set
...
-      alt="Pauline Clayton"
-      className="c1"
-      height={null}
-      sizes={null}
-      src="https://ichef.bbci.co.uk/news/640/cpsprodpb/E7DB/production/_101655395_paulineclayton.jpg"
-      srcSet="https://ichef.bbci.co.uk/news/640/cpsprodpb/E7DB/production/_101655395_paulineclayton.jpg 640w"
-      width={640}
-    />
-  </noscript>
+  <noscript />
```
Comment (dr3) [inline src/app/containers/ImageWithPlaceholder/__snapshots__/index.test.jsx.snap:46]: Snapshot no longer reflects previous coverage, needs investigating

File: src/app/containers/PageHandlers/withPageWrapper/__snapshots__/index.test.jsx.snap [hunk 9]
```diff
@@ -46,491 +45,300 @@ Array [
...
-        </p>
-      </div>
+          BBC News
+        </span>
+      </a>
     </div>
-  </footer>,
-]
+  </div>
+</header>
```
Comment (dr3) [inline src/app/containers/PageHandlers/withPageWrapper/__snapshots__/index.test.jsx.snap]: Snapshot no longer reflects previous coverage, needs investigating

**Review by dr3 — CHANGES_REQUESTED** [review]
Code looks good, but some of the new snapshots dont reflect the info in the old ones, need to investigate why and how we could bring it back :)

**Comment by FK78** [comment]
Blocked by #[1917](https://github.com/bbc/psammead/pull/1917/files). Closed till further notice.

File: src/app/components/IfAboveIE9Comment/__snapshots__/index.test.jsx.snap [hunk 10]
```diff
@@ -1,23 +1,7 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`Timestamp should render if not IE9 comment 1`] = `
```
Comment (jroebu14) [inline src/app/components/IfAboveIE9Comment/__snapshots__/index.test.jsx.snap]: Timestamp text here is a little bit misleading https://github.com/bbc/simorgh/blob/6586a197a668233cf49b30df7ecd506396e6dc02/src/app/components/IfAboveIE9Comment/index.test.jsx#L5

**Comment by FK78** [comment]
Snapshot coverage issue may be resolved by PR #[1958](https://github.com/bbc/psammead/pull/1958)

File: src/app/components/Metadata/__snapshots__/index.test.jsx.snap [hunk 11]
```diff
@@ -1,1623 +1,15 @@
...
-    rel="apple-touch-startup-image"
-  />
-  <link
-    href="/favicon.ico"
-    rel="shortcut icon"
-    type="image/x-icon"
-  />
-</HelmetWrapper>
-`;
+exports[`Metadata News AMP article should render correctly 1`] = `<div />`;
```
Comment (FK78) [inline src/app/components/Metadata/__snapshots__/index.test.jsx.snap]: Snapshot coverage may be restored after PR #[3694](https://github.com/bbc/simorgh/pull/3694)

File: src/app/components/LinkedData/__snapshots__/index.test.jsx.snap [hunk 12]
```diff
@@ -1,40 +1,7 @@
...
-  encodeSpecialCharacters={true}
->
-  <script
-    type="application/ld+json"
-  >
-    {"@context":"http://schema.org","@type":"WebPage","url":"https://www.bbc.com/igbo","publisher":{"@type":"NewsMediaOrganization","name":"BBC News","publishingPrinciples":"https://www.bbc.com/news/help-41670342","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://news.files.bbci.co.uk/ws/img/logos/og/igbo.png"}},"image":{"@type":"ImageObject","width":1024,"height":576,"url":"https://news.files.bbci.co.uk/ws/img/logos/og/igbo.png"},"thumbnailUrl":"https://news.files.bbci.co.uk/ws/img/logos/og/igbo.png","mainEntityOfPage":{"@type":"WebPage","@id":"https://www.bbc.com/igbo","name":"Ogbako"}}
-  </script>
-</HelmetWrapper>
-`;
+exports[`LinkData should correctly render metadata for frontpages 1`] = `<div />`;
```
Comment (FK78) [inline src/app/components/LinkedData/__snapshots__/index.test.jsx.snap:3]: Once PR #3694 is merged, snapshot coverage might be fixed.

File: src/app/components/Metadata/__snapshots__/index.test.jsx.snap (hunk 11 above)
Comment (FK78) [inline src/app/components/Metadata/__snapshots__/index.test.jsx.snap:3]: Once PR #3694 is merged, snapshot coverage might be fixed.

File: src/app/Layouts/defaultPageWrapper.test.jsx [hunk 13]
```diff
@@ -1,14 +1,31 @@
...
+import { ServiceContextProvider } from '../contexts/ServiceContext';
+import { ToggleContext } from '../contexts/ToggleContext';
+import { RequestContext } from '../contexts/RequestContext';
 
 describe('defaultPageWrapper', () => {
   const propsWithChildren = {
     children: <h2>Child element</h2>,
   };
 
-  shouldShallowMatchSnapshot(
```
Comment (dr3) [inline src/app/Layouts/defaultPageWrapper.test.jsx:10]: With this change you can delete
```
<GlobalStyle />
<ServiceWorkerContainer />
<ManifestContainer />
<MPulseBeacon />
```
In `simorgh/src/app/Layouts/defaultPageWrapper.jsx` and it still passes, can you add assertion tests to ensure we maintain coverage please?

File: src/app/components/IfAboveIE9Comment/index.test.jsx [hunk 14]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
-import { shouldMatchSnapshot } from '../../../testHelpers';
+import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
 import IfAboveIE9 from '.';
 
-describe('Timestamp', () => {
+describe('Children', () => {
```
Comment (dr3) [inline src/app/components/IfAboveIE9Comment/index.test.jsx]:
```suggestion
describe('IfAboveIE9Comment', () => {
```

File: src/app/components/LinkedData/__snapshots__/index.test.jsx.snap (hunk 12 above)
Comment (dr3) [reply src/app/components/LinkedData/__snapshots__/index.test.jsx.snap:3]: We should block this PR on that then :), lots of lost coverage

File: src/app/containers/ATIAnalytics/canonical/__snapshots__/index.test.jsx.snap [hunk 15]
```diff
@@ -1,17 +1,5 @@
...
-    style={
-      Object {
-        "position": "absolute",
-      }
-    }
-    width="1px"
-  />
-</noscript>
-`;
+exports[`Canonical ATI Analytics should render a noscript image for non-JS users: <img height="1px" width="1px" alt="" style="position:absolute" src="https://foobar.com?key=value&amp;key2=value2"/> 1`] = `undefined`;
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/canonical/__snapshots__/index.test.jsx.snap:3]: Im not sure this is working, whats the `undefined`?

File: src/app/containers/ATIAnalytics/canonical/index.test.jsx [hunk 16]
```diff
@@ -29,11 +29,11 @@ describe('Canonical ATI Analytics', () => {
   });
 
   it('should render a noscript image for non-JS users', () => {
-    const renderedATI = render(
+    const { renderedATI } = render(
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/canonical/index.test.jsx:32]: Not sure this key is right, i think it needs to be something that `render` provides?

File: src/app/containers/ATIAnalytics/canonical/index.test.jsx [hunk 17]
```diff
@@ -29,11 +29,11 @@ describe('Canonical ATI Analytics', () => {
...
       <CanonicalATIAnalytics pageviewParams={mockPageviewParams} />,
     );
 
-    expect(renderedATI.html()).toBe(
+    expect(renderedATI).toMatchSnapshot(
       `<img height="1px" width="1px" alt="" style="position:absolute" src="https://foobar.com?key=value&amp;key2=value2"/>`,
```
Comment (dr3) [inline src/app/containers/ATIAnalytics/canonical/index.test.jsx:37]: Whats this doing? Im not sure these are the same

File: src/app/containers/App/App.test.jsx [hunk 18]
```diff
@@ -57,6 +57,34 @@ describe('App', () => {
     expect(wrapper).toMatchSnapshot();
   });
 
+  //   // Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (dr3) [inline src/app/containers/App/App.test.jsx:60]: Commented out?

File: src/app/containers/App/index.jsx [hunk 19]
```diff
@@ -22,3 +22,5 @@ export const ServerApp = props => (
     />
   </StaticRouter>
 );
+
+//assertion testing and remove snapshots and test props being passed in
```
Comment (dr3) [inline src/app/containers/App/index.jsx:26]: Whats this?

File: src/app/containers/App/__snapshots__/index.test.jsx.snap [hunk 20]
```diff
@@ -1,69 +0,0 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
-
-exports[`ClientApp should render correctly 1`] = `
```
Comment (dr3) [inline src/app/containers/App/__snapshots__/index.test.jsx.snap:3]: File deleted?

File: src/app/containers/App/index.test.jsx [hunk 21]
```diff
@@ -3,36 +3,68 @@
...
-  );
+  it('should render correctly', () => {
+    render(<ClientApp data="someData!" routes={['someRoute']} />);
+    expect(App).toHaveBeenCalledWith(
+      { initialData: 'someData!', routes: ['someRoute'] },
+      {},
+    );
+    expect(BrowserRouter).toHaveBeenCalledWith(
+      {
+        children: expect.anything(),
```
Comment (dr3) [inline src/app/containers/App/index.test.jsx:30]: We need to find an alternative to `expect.anything()`, happy to help find one :)

File: src/app/containers/ArticleFigure/__snapshots__/index.test.jsx.snap (hunk 7 above)
Comment (dr3) [reply src/app/containers/ArticleFigure/__snapshots__/index.test.jsx.snap:103]: +1

File: src/app/containers/Blocks/index.test.jsx [hunk 22]
```diff
@@ -75,7 +75,7 @@ describe('Blocks', () => {
   describe('with unknown blocks', () => {
     const unknownBlocks = [unknownTestBlock, parentBlockNotAllowed];
 
-    shouldShallowMatchSnapshot(
+    shouldMatchSnapshot(
```
Comment (dr3) [inline src/app/containers/Blocks/index.test.jsx:78]: Should this be a `isNull`?

File: src/app/containers/Blocks/__snapshots__/index.test.jsx.snap [hunk 23]
```diff
@@ -1,200 +1,15 @@
...
-        },
-      ]
-    }
-    type="text"
-    typeOfPreviousBlock="subheadline"
-  />,
-  null,
-  null,
-]
+<p>
```
Comment (dr3) [inline src/app/containers/Blocks/__snapshots__/index.test.jsx.snap:10]: Lots of stuff missing here, any ideas why Its not rendering stuff like `This is a headline!`?

File: src/app/containers/Footer/__snapshots__/index.test.jsx.snap [hunk 24]
```diff
@@ -311,4 +311,4 @@ exports[`FooterContainer should render correctly 1`] = `
 </footer>
 `;
 
-exports[`FooterContainer should render null when footer config not availible 1`] = `null`;
+exports[`FooterContainer should render null when footer config not availible 1`] = `<div />`;
```
Comment (dr3) [inline src/app/containers/Footer/__snapshots__/index.test.jsx.snap:314]: Does this need an isNull check?

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/containers/App/index.jsx (hunk 19 above)
Comment (FK78) [reply src/app/containers/App/index.jsx:26]: Notes I've left in by accident.

**Comment by FK78** [comment]
PR is being split up so it is easier to review.

### Small changes
- #3708
- #3776
- #3814

### Major changes
- #3711
- #3842
- #3854
- #3886
- #3945
- #4058

**Pull request closed without merging**